packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/private/utils.ts

```diff
@@ -0,0 +1,60 @@
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core/';
+import { Construct } from 'constructs';
+
+/**
+ * Generate policy resource for a list of build ARNs or report ARNs.
+ */
+export function generatePolicyResource(scope: Construct, ids: string[], service: string, resource: string): string[] {
+  let resources = new Set();
+  let resourceArns = [];
+
+  if (sfn.JsonPath.isEncodedJsonPathStringList(ids)) {
+    resourceArns.push(generateArn(scope, service, resource, '*'));
+    return resourceArns;
+  }
+
+  for (let id of ids) {
+    let resourceName = cdk.Stack.of(scope).parseArn(id).resourceName;
+    if (resourceName) {
+      resources.add(resourceName.split(':')[0]);
+    }
+  }
+
+  for (let res of resources) {
```

Comment:
do all of these `let` statements need to be `let` or can they be `const` as they're not really re-assigned?
applies a few times in this file